### ISSUE TITLE
next song and random

### ISSUE THREAD

**Issue by jeremya37** [tizonia/tizonia-openmax-il#590]
Hello,

Is it possible to have keyboard shortcuts in the command line?

That is, type something like: tizonia -n to put the next song?

second thing, when you want a playlist given the list and always in the same order regardless of the listening platform. Is there a way that the chonson is totally random at each reading?

Thank you in advance

Sorry for my English I am French.

versions

tizonia 0.18.0.

**Comment by Psycoguana** [comment]
For the next song type `n` while playing something.
For shuffle type something like: `tizonia --soundcloud-user-playlist some_playlist -s`, -s being the shuffle option.

For more commands type : `tizonia --help keyboard` and `tizonia --help global`

**Comment by jeremya37** [comment]
thank you but the problem is that I would like to order in order to integrate domotic ...

-s ist perfect !!!

**Comment by jofftiquez** [comment]
This should be on the documentation.

**Comment by jeremya37** [comment]
In which documentation? because I can not find anything

**Comment by tizonia** [comment]
Hi all!,

thanks for the feedback.

With regards to the documentation, have you guys looked through the docs at http://docs.tizonia.org/? The documentation is very basic but there are some screenshots that should be helpful. E.g.:
- http://docs.tizonia.org/usage/global.html
- http://docs.tizonia.org/usage/keyboard.html

Contributions are more than welcome, If anybody feels like helping with the docs, let me know. Doc sources are located in https://github.com/tizonia/tizonia-openmax-il/tree/master/docs

@jeremya37 
>  I would like to order in order to integrate domotic ...

At the moment Tizonia is currently all about automatically generated playlists provided by the various services, or user playlists that are located in the cloud. For example, you can create your own playlist in Spotify, Google Play Music, SoundCloud or YouTube and get those played back with Tizonia. However, you can't feed local playlists to Tizonia in either pls or m3u formats. There is no support for that right now. If you are interested, please feel free to raise a separate issue with your suggestion.

**Comment by jeremya37** [comment]
ok i am just looking for a command to type in the terminal to go to the next music. To replace the keyboard -n

**Comment by tizonia** [comment]
@jeremya37  Sorry for the delayed reply. 

Are you saying that you would like to use some command on *another* terminal, different from where Tizonia is running? Tizonia supports MPRISv2 and you could control Tizonia from another terminal using the remote control script. I've never released it in the debs or Snap package out of lack of time, but you can download it from here:

https://github.com/tizonia/tizonia-openmax-il/blob/master/player/tools/tizonia-remote.in

Just grab it, make it executable, and remember to enable MPRIS in your configuration file:
https://github.com/tizonia/tizonia-openmax-il/blob/9004bc40b89eeafb04b28fbb2b772e47dd62fdc9/config/src/tizonia.conf.in#L63

**Comment by jeremya37** [comment]
oui voila je voudrais que mon assistant domotique lance une playlist en arrière plan ( ce qu'il fait déjà très bien).

Et que sur un terminal en ssh je puisse mettre en pause muter ou mettre la chanson suivante en ligne de commande. ( ex tizonia cmdnext)

Je vais me pencher sur ce que tu m'as donné mais je ne m'y connais pas de trop et quand j'active MPRIS j'ai une erreur 'Tiz::DBus::Error'

**Comment by jeremya37** [comment]
> Just grab it, make it executable

what do you mean by: 

> Just grab it, make it executable


I'm starting out sorry

**Comment by tizonia** [comment]
@jeremya37 

Yes, sorry I should've explained. The tool is called tizonia-remote. It is a shell script. You can download it directly from the terminal using wget or similar, and then you can make it executable and put it somewhere in your path. 

```
$ wget https://raw.githubusercontent.com/tizonia/tizonia-openmax-il/master/player/tools/tizonia-remote.in
$ mv tizonia-remote.in tizonia-remote
$ chmod +x tizonia-remote
$ sudo install tizonia-remote /usr/local/bin 
```

**Comment by jeremya37** [comment]
thank you so much
So if I understand correctly reading the command is launched like this:

` tizonia-remote break`

?

By cons I still have the same error when I put `mpris-enabled = true` in both conf files.

> 
> 0 | gladys | terminated after throwing an instance of 'Tiz :: DBus :: Error'
> 0 | gladys | what (): Unable to autolaunch a dbus-daemon without a $ DISPLAY for X11
> 0 | gladys | Aborted (core dumped)
> 0 | gladys | at ChildProcess.exithandler (child_process.js: 275: 12)
> 0 | gladys | at emitTwo (events.js: 126: 13)
> 0 | gladys | at ChildProcess.emit (events.js: 214: 7)
> 0 | gladys | at maybeClose (internal / child_process.js: 925: 16)
> 0 | gladys | at Socket.stream.socket.on (internal / child_process.js: 346: 11)
> 0 | gladys | at emitOne (events.js: 116: 13)
> 0 | gladys | at Socket.emit (events.js: 211: 7)
> 0 | gladys | at Pipe._handle.close [as _onclose] (net.js: 554: 12)

**Comment by tizonia** [comment]
> enabled mpris mpris-enabled = true in the configuration file:
```
sudo nano /etc/xdg/tizonia/tizonia.conf 
```
I think you probably should modify the user's configuration file, which is located in $HOME/.config/tizonia/tizonia.conf

With regards to running tizonia-remote on a headless system, you will need to make sure that D-BUS is available on your system (you need to install it first using the package manager).

Once the D-Bus daemon is installed, you need to launch it at the start of your user's session. There may be many ways of doing this, but the following shell snippet works for me. I usually add this to my $HOME/.bashrc or $HOME/.zshrc  


```
DBUS_VARS_FILE="$HOME/.dbus_vars" # Where file with variables is stored
# This function launches dbus and saves environment variables
launch_dbus() {
    OLDUMASK=$(umask) # Save umask
    umask 0077 # Set secure umask
    if dbus-launch --auto-syntax > "$DBUS_VARS_FILE"; # Launch DBus and save its environment variables
    then
        . "$DBUS_VARS_FILE" # Load environment variables
    else
        # Error
        echo Failed to launch DBus
    fi
    umask $OLDUMASK # Restore umask
}


    if [ -f "$DBUS_VARS_FILE" ];
    then
        . "$DBUS_VARS_FILE" # Load environment variables
        if ps ax | grep -q '^ '"$DBUS_SESSION_BUS_PID"; # Check whether DBus is running
        then
            echo -n # DBus is running, so do nothing
        else
            launch_dbus # DBus is not running (old file was loaded)
        fi

    else
        launch_dbus
    fi
```

**Comment by jeremya37** [comment]
I can not find anything there .... I even install dbus I always have this error:


terminated after throwing an instance of 'Tiz :: DBus :: Error'
  what (): Unable to autolaunch a dbus-daemon without a $ DISPLAY for X11
Aborted



So, thank you very much for your patience I will not continue on the creation of plugins too bad but I keep tizonia for my home automation anyway

**Comment by antoinebou13** [comment]
@jeremya37
I reproduce the issue in the my Windows Subsystem for Linux 
`sudo apt install dbus-x11`
and i install xming on windows to have a x11 server display and i added
`export DISPLAY=:0`
and since that i don't have the issue

`terminate called after throwing an instance of 'Tiz::DBus::Error'
  what():  /usr/bin/dbus-launch terminated abnormally with the following error: Autolaunch error: X11 initialization failed.

Aborted (core dumped)`

,but the remote application doesn't work for me

`[Spotify] [Connected] : '7yWsPfsWneO9DjvcoXOs07'.
shared memfd open() failed: Function not implemented`

tizonia-remote pause
`D-Bus per-session daemon address is: unix:abstract=/tmp/dbus-gMIOHKFNT2,guid=08f331482c69365fe3f749045ced86fe
D-Bus per-session daemon PID is    : 29519`

doesn't know if that help you problem because you have a displayless device
